daemon/container_linux.go

```diff
@@ -513,13 +513,21 @@ func (container *Container) buildPortMapInfo(n libnetwork.Network, ep libnetwork
 		networkSettings.MacAddress = mac.(net.HardwareAddr).String()
 	}
 
+	networkSettings.Ports = nat.PortMap{}
+
+	if expData, ok := driverInfo[netlabel.ExposedPorts]; ok {
+		for _, tp := range expData.([]types.TransportPort) {
```

Comment:
As far as I can tell it's something that the driver builds: I don't think we can trust a driver to always give us something of the expected type. But maybe I'm being paranoid.